src/results.cpp

```diff
@@ -370,20 +374,24 @@ util::Optional<Mixed> Results::aggregate(size_t column, bool return_none_for_emp
 
 util::Optional<Mixed> Results::max(size_t column)
 {
-    return aggregate(column, true, "max",
-                     [=](auto const& table) { return table.maximum_int(column); },
-                     [=](auto const& table) { return table.maximum_float(column); },
-                     [=](auto const& table) { return table.maximum_double(column); },
-                     [=](auto const& table) { return table.maximum_timestamp(column); });
+    size_t return_ndx;
+    auto results = aggregate(column, true, "max",
+                     [&](auto const& table) { return table.maximum_int(column, &return_ndx); },
+                     [&](auto const& table) { return table.maximum_float(column, &return_ndx); },
+                     [&](auto const& table) { return table.maximum_double(column, &return_ndx); },
+                     [&](auto const& table) { return table.maximum_timestamp(column, &return_ndx); });
+    return return_ndx == npos ? none : results;
```

Comment:
We could push this repeated test into `aggregate` since it is of the same vein as the `return_none_for_empty` work that it already handles. It'd require adding a `return_ndx` reference parameter to each of these lambdas though, so maybe that work would outweigh the benefit from removing the small amount of repetition?